src/extension.ts

```diff
@@ -249,6 +259,53 @@ export async function attach(
     } while (useRetry && Date.now() - startTime < timeout);
 }
 
+export async function attachToCurrentDebugTarget(context: vscode.ExtensionContext): Promise<void> {
```

Comment:
I think is better to have the logic described here in it's own class and factory (CDPDebugTarget && CDPDebugTargetProvider), that will help with encapsulation.

Reply:
I feel like that might be overkill? the cdpTarget and cdpTargetsprovider need to handle finding and providing several targets available from the browser of different type, and handles the sidebar ui, but this logic is the same every time, and is shorter than the launch/attach workflow, and has no UI components. I would basically just be moving this function into a separate file, but would still need to call/register it here

Reply:
my bad, forgot to delete this one after the clarification on the other comment 👍 (I thought we were handling the connection through different sockets)